eth/downloader/downloader_test.go

```diff
@@ -769,8 +759,8 @@ func testThrottling(t *testing.T, protocol int, mode SyncMode) {
 			}
 			frozen = int(atomic.LoadUint32(&blocked))
 			retrieved = len(tester.ownBlocks)
-			tester.downloader.queue.lock.RUnlock()
-			tester.lock.RUnlock()
+			tester.downloader.queue.lock.Unlock()
+			tester.lock.Unlock()
```

Comment:
The queue lock is now a `sync.Mutex`. 
I don't have a proof, but it looked like `sync.RWMutex` was not appropriate.
Read/write locks can boost performance when there are many reads and occasional
writes. This is not the case here, all fetch operations update the queue all the time.
If you want it to be changed back, I can oblige.